torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2463,6 +2463,7 @@ def _init(
         )
 
         self.guards: List[ShapeGuard] = []
+        self.input_guards: List[ShapeGuard] = []
```

Comment:
i ended up separating these because symbolic shape recording pass validation was failing otherwise

Reply:
cc @ysiraichi

Reply:
This seems like a dorky way to do this. One superficial complaint is that ShapeEnv is not Inductor specific, and yet we have named this thing "input guards", what the heck is an input guard?

For me, an intuitive code organization is that I want some way to call `produce_guards` with an explicit guards list, rather than using the guards that is accumulated directly on the ShapeEnv. That seems like a very modest change and doesn't require adding an extra field on ShapeEnv.

You need to make sure to not run translation validation on this `produce_guards` call either, as it will fail (and for good reason). TV should only run when you don't explicitly pass in guards to produce.